lib/Driver/ToolChains/Cuda.cpp

```diff
@@ -830,3 +830,42 @@ VersionTuple CudaToolChain::computeMSVCVersion(const Driver *D,
                                                const ArgList &Args) const {
   return HostTC.computeMSVCVersion(D, Args);
 }
+
+static void AddFlangSysIncludeArg(const ArgList &DriverArgs,
+                                  ArgStringList &Flang1Args,
+                                  ToolChain::path_list IncludePathList) {
+  std::string ArgValue; // Path argument value
+
+  // Make up argument value consisting of paths separated by colons
+  bool first = true;
+  for (auto P : IncludePathList) {
+    if (first) {
+      first = false;
+    } else {
+      ArgValue += ":";
+    }
+    ArgValue += P;
+  }
+
+  // Add the argument
+  Flang1Args.push_back("-stdinc");
+  Flang1Args.push_back(DriverArgs.MakeArgString(ArgValue));
+}
+
+void CudaToolChain::AddFlangSystemIncludeArgs(const llvm::opt::ArgList &DriverArgs,
+                                              llvm::opt::ArgStringList &Flang1Args) const {
+  path_list IncludePathList;
+  const Driver &D = getDriver();
+
+
+  if (DriverArgs.hasArg(options::OPT_nostdinc))
+    return;
+
+  {
+    SmallString<128> P(D.InstalledDir);
+    llvm::sys::path::append(P, "../include");
+    IncludePathList.push_back(P.str());
+  }
+
+  AddFlangSysIncludeArg(DriverArgs, Flang1Args, IncludePathList);
+}
```

Comment:
OK. I see that this is essentially just block copied from the earlier diffs to Linux.cpp. Does Linux::AddFlangSystemIncludeArgs() get called? Is it called first?

Reply:
The Linux::AddFlangSystemIncludeArgs() gets called first compilation, which is for the host code. The clang driver starts compilation another time to compile device code. As I request nvidia target for OpenMP implementation, it uses cuda toolchain. In this time, CudaToolChain::AddFlangSystemIncludeArgs gets called.